pkg/kubelet/dockertools/kube_docker_client.go

```diff
@@ -303,6 +396,18 @@ func parseDockerTimestamp(s string) (time.Time, error) {
 	return time.Parse(time.RFC3339Nano, s)
 }
 
+func getDefaultContext() (context.Context, context.CancelFunc) {
+	return context.WithTimeout(context.Background(), defaultTimeout)
+}
+
+// contextError checks the context, and returns error if the context is timeout.
+func contextError(ctx context.Context) error {
+	if ctx.Err() == context.DeadlineExceeded {
```

Comment:
Another error is `var Canceled = errors.New("context canceled")`.
We never cancel a context, so that should not happen. But I agree that we should still expose the other potential context error just in case.